pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -440,6 +440,51 @@ def remove_cpds(self, *cpds):
                 cpd = self.get_cpds(cpd)
             self.cpds.remove(cpd)
 
+    def get_cardinality(self, node=None):
+        """
+        Returns the cardinality of the node.
+
+        Parameter
+        ---------
+        node: tuple (node_name, time_slice)
+            The node should be in the following form (node_name, time_slice).
+            Here, node_name is the node that is inserted while the time_slice is
+            an integer value, which denotes the index of the time_slice that the
+            node belongs to.
+
+        Returns
+        -------
+        int or dict : If node is specified returns the cardinality of the node.
+                      If node is not specified returns a dictionary with the given
+                      variable as keys and their respective cardinality as values.
+
+        Examples:
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> from pgmpy.factors.discrete import TabularCPD
+        >>> dbn = DBN()
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('D',0),('D',1)),(('I',0),('I',1))])
+        >>> grade_cpd =  TabularCPD(('G',0), 3, [[0.3,0.05,0.9,0.5],
+        ...                                      [0.4,0.25,0.8,0.03],
+        ...                                      [0.3,0.7,0.02,0.2]], [('I', 0),('D', 0)],[2,2])
+        >>> dbn.add_cpds(grade_cpd)
+        >>> dbn.get_cardinality(('D',0))
+        2
+        >>> dbn.get_cardinality()
+        defaultdict(int, {('D', 0): 2, ('D', 1): 2, ('G', 0): 3, ('I', 0): 2, ('I', 1): 2})
+        """
+        if node:
+            if node not in super(DynamicBayesianNetwork, self).nodes():
+                raise ValueError('Node not present in the model.')
+            else:
+                temp_node = (node[0], 1 - node[1]) if node[1] else node
```

Comment:
@lohani2280 I am unable to understand why you are using `1 - node[1]`. Could you please explain a bit ?

Reply:
@ankurankan 
```
        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
        >>> from pgmpy.factors.discrete import TabularCPD
        >>> dbn = DBN()
        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('D',0),('D',1)),(('I',0),('I',1))])
        >>> grade_cpd =  TabularCPD(('G',0), 3, [[0.3,0.05,0.9,0.5],
        ...                                      [0.4,0.25,0.8,0.03],
        ...                                      [0.3,0.7,0.02,0.2]], [('I', 0),('D', 0)],[2,2])
        >>> dbn.add_cpds(grade_cpd)
``` 
In DBN the cardinality of the node will be same at every time slice. Now, If the node queried  using `get_cardinality` is in time slice = 0 then `temp_node = node`. However, if the queried node is not in time slice 0 then using `1-node[1]` ,i brought it to time slice 0 and then proceed to output the cardinality of the queired node at time slice =0. 

For example - 
If `dbn.get_cardinality(('D',1))` is called then -
`node = ('D',1)`
`node[1] = 1`
therefore, `temp_node = (node[0],1 - node[1]) = ('D',0)` 

Similarly if   `dbn.get_cardinality(('D',0))` is called then -
`node = ('D',0)`
`node[1] = 0`
therefore, `temp_node = node = ('D',0)`.

Actually , with this i am thinking to replace https://github.com/pgmpy/pgmpy/blob/0f62927e6404742a73ea1796587e345b74100775/pgmpy/models/DynamicBayesianNetwork.py#L512 with following piece of code to solve issue https://github.com/pgmpy/pgmpy/issues/877
```
            parents_card = []
            for parent in parents:
                parents_card.append(self.get_cardinality(parent))
```

Reply:
@lohani2280 But what if someone queries in let's say time slice 5, with the node `('D', 5)`. This would fail in that case. Right ?

Reply:
@ankurankan Ohh!! sorry my bad. I went out with the flow of the example in the docstring.